Changed ExpandIcon to accept default state.

dragostis opened the pull request:
Changed the ExpandIcon constructor to take a default expansion state.
If the widget gets rebuilt with a different expansion value, the
animation will trigger but the callback will not.

Hixie commented on packages/flutter/lib/src/material/expand_icon.dart:
```diff
@@ -75,6 +84,19 @@ class _ExpandIconState extends State<ExpandIcon> {
+
   void _handlePressed() {
     setState(() {
       _isExpanded = !_isExpanded;
```
you should be able to remove this setState section

HansMuller commented on packages/flutter/lib/src/material/expand_icon.dart:
```diff
@@ -23,21 +23,29 @@ class ExpandIcon extends StatefulWidget {
+  /// Whether the icon is in an expanded state.
+  ///
+  /// Rebuilding the widget with a different [isExpanded] value will trigger
+  /// the animation, but will not trigger the callback.
```
It would be good to say 'the onPressed callback'.

HansMuller commented:
Nice change! LGTM

dragostis committed:
Changed ExpandIcon to accept default state.

Changed the ExpandIcon constructor to take a default expansion state.
If the widget gets rebuilt with a different expansion value, the
animation will trigger but the callback will not.
